src/v/cluster/topic_table.cc

```diff
@@ -878,6 +787,105 @@ std::vector<model::ntp> topic_table::all_updates_in_progress() const {
     return ret;
 }
 
+void topic_table::change_partition_replicas(
+  model::ntp ntp,
+  const std::vector<model::broker_shard>& new_assignment,
+  topic_metadata_item& metadata,
+  model::offset o) {
+    auto current_assignment_it = metadata.get_assignments().find(
```

Comment:
can we pass the current_assignment replicas as a reference from the caller ? This way we will not have to do a lookup here

Reply:
`current_assignment.id` is not required as we already have `ntp`